File: techila/examples/python/Features/file_handler/run_filehandler.py
# Copyright 2012-2016 Techila Technologies Ltd.

# This script contains the Local Control Code, which will be used to create the
# computational Project and post-process the output files.
#
# Usage:
# execfile('run_filehandler.py')
# run_filehandler()

# Import the techila package
import techila

# Load other necessary packages
import os
import shutil

# This function is the filehandler function, which will be
# used to post-process the output files. Function will be called once for
# each output file.
def filehandler_func(file):
  # Display the location of the file on the End-Users computer
  print(file)

  # Display contents of the file
  f = open(file, 'r')
  line = f.readline()
  print(line)
  f.close()

  # Copy the file to the current working directory
  filename = os.path.basename(file)
  shutil.copy(file, os.getcwd())

# This function contains the peach function call, which will be
# used to create the computational Project
def run_filehandler():
    jobs = 5 # Will be used to set the number of Jobs to 5

    result = techila.peach(funcname = 'worker_dist', # Function that will be called on Workers
                           files = 'worker_dist.py', # Files that will be sourced on Workers
                           params = ['<param>'], # Input parameters for the executable function
                           peachvector = range(1, jobs+1), # Set the number of Jobs to 5
                           outputfiles = ['output_file.*;regex=1'], # Files to returned from Workers
                           filehandler = filehandler_func, # Name of the filehandler function
                           )

File: techila/examples/R/Features/file_handler/run_filehandler.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This script contains the Local Control Code, which will create the
# computational Project.
#
# Usage:
# source("run_filehandler.r")
# run_filehandler()
#
# Example:
# run_filehandler()

# Load the techila library
library(techila)


# This function contains the filehandler function, which will be
# called once for each result file received.
filehandler_func <- function(file) {

  # Display the location of the result file on the End-Users computer
  print(file)

  # Load contents of the file to memory
  load(file)
  if (exists("sample1")) {  # Current file is 'file1'
    print(sample1)
  }
  if (exists("sample2")) {  # Current file is 'file2'
    print(sample2)
  }
}

# This function contains the peach function call, which will be
# used to create the computational Project
run_filehandler <- function() {

  result <- peach(funcname = "worker_dist", # Function that will be called on Workers
                  files = "worker_dist.r", # Files that will be sourced on Workers
                  params = list("<param>"), # Input parameters for the executable function
                  peachvector = 1:2, # Set the number of Jobs to two (2)
                  outputfiles = list("file1", "file2"), # Files to returned from Workers
                  filehandler = filehandler_func, # Name of the filehandler function
                  sdkroot = "../../../..", # Location of the techila_settings.ini file
                 )
}
